lighthouse-core/gather/gather-runner.js

```diff
@@ -187,17 +206,56 @@ class GatherRunner {
       const status = `Retrieving: ${gatherer.name}`;
       return chain.then(_ => {
         log.log('status', status);
-        return gatherer.afterPass(options, passData);
-      }).then(ret => {
+        const artifactPromise = Promise.resolve().then(_ => gatherer.afterPass(options, passData));
+        gathererResults[gatherer.name].push(artifactPromise);
+        return artifactPromise.catch(err => {
+          if (!err.recoverable) {
```

Comment:
that's a fair point. Ideally recoverable errors will be a rare case, but it is a bit awkward to create them (see the tests in this PR, for instance). We could easily make a `RecoverableError` subclass that folks could use if they want, which would make that a lot cleaner.

Both versions could still work, the important bit being the `recoverable` property. That would let a gatherer catch an error it might expect sometimes happens, add a `err.recoverable = true` and then re-throw. Would having both work be confusing for folks writing gatherers, though?